client/client.go

```diff
@@ -280,6 +281,49 @@ func newHttpClient() *http.Client {
 	return &http.Client{}
 }
 
+func dialOpenSSL(conf Config, network string, addr string) (net.Conn, error) {
+	contextSSL, err := openssl.NewCtx()
+	// probably should consider reusing the context, but then we
+	// have to propagate it with every request.
+	err = contextSSL.LoadVerifyLocations(conf.ServerCert, "")
+	if err != nil {
+		return nil, err
+	}
+	flags := openssl.DialFlags(0)
+
+	if conf.NoVerify {
+		flags = openssl.InsecureSkipHostVerification
+	}
+
+	conn, err := openssl.Dial("tcp", addr, contextSSL, flags)
+	if conn == nil || err != nil {
+		return nil, err
+	}
```

Comment:
Mmm. could we have `conn == nil` and `err == nil`? In that case the return would not make sense, we would need two different ones:
```suggestion
	conn, err := openssl.Dial("tcp", addr, contextSSL, flags)
	if err != nil {
		return nil, err
	}
	if con == nil {
	        ...
	}
```

Reply:
> Mmm. could we have `conn == nil` and `err == nil`? In that case the return would not make sense, we would need two different ones:

no chance for that. I checked that every time [DialSession](https://github.com/spacemonkeygo/openssl/blob/master/net.go#L96) returns only `nil,err` (where err!=nil) or `conn,nil` (where conn!=nil)

Reply:
Then I would only check for `err`. The if condition is somewhat dangerous as-is.

Reply:
will do! thanks.

Reply:
fixed.